.github/actions/check-image-and-changes/action.yaml

```
@@ -0,0 +1,57 @@
+name: "Check Image and Relevant Changes"
+description: "Checks if the image already exists and if changes are relevant"
+
+outputs:
+  image_exists:
+    description: "True if the image already exists"
+    value: ${{ steps.image-check.outputs.exists }}
+  relevant_changes:
+    description: "True if changes require a build"
+    value: ${{ steps.changes.outputs.relevant }}
+  short_sha:
+    description: "Short SHA of the latest commit"
+    value: ${{ steps.get-sha.outputs.short_sha }}
+
+runs:
+  using: "composite"
+  steps:
+    - name: Get the last commit short SHA of the PR
+      id: get-sha
+      uses: ./.github/actions/get-sha  # Chamando a action existente
+
+    - name: Check if Image Already Exists
+      id: image-check
+      shell: bash
+      run: |
+        IMAGE_TAG_COMMIT="pr-${{ github.event.number }}-${{ steps.get-sha.outputs.short_sha }}"
```

since the get-sha action commits an env var, you can just use...

```suggestion
        IMAGE_TAG_COMMIT="pr-${{ github.event.number }}-${{ env.SHORT_SHA }}"
```